openadapt/record.py

```diff
@@ -618,12 +634,30 @@ def record(
 
     # TODO: discard events until everything is ready
 
+    collect_stats()
+
     try:
         while True:
             time.sleep(1)
     except KeyboardInterrupt:
         terminate_event.set()
 
+    collect_stats()
+
+    stats = snapshots[-1].compare_to(snapshots[-2], 'traceback')
+
+    for stat in stats[:3]:
+        print(
+            "{} new KiB {} total KiB {} new {} total memory blocks: ".format(
+                stat.size_diff / 1024,
+                stat.size / 1024,
+                stat.count_diff,
+                stat.count))
```

Comment:
What do you think about:
```
        new_KiB = stat.size_diff / 1024
        total_KiB = stat.size / 1024
        new_blocks = stat.count_diff
        total_blocks = stat.count
        logger.info(f"{new_KiB=} {total_KiB=} {new_blocks=} {total_blocks=}")
```

Reply:
```
    for stat in stats[:3]:
        new_KiB = stat.size_diff / 1024
        total_KiB = stat.size / 1024
        new_blocks = stat.count_diff
        total_blocks = stat.count
        source = stat.traceback.format()[0]
        logger.info(f"{new_KiB=} {total_KiB=} {new_blocks=} {total_blocks=} {source=}")
```

output:
![image](https://github.com/MLDSAI/OpenAdapt/assets/88949118/da1f7493-bd70-4dd8-b4e6-a5d676c23e31)